src/connections/XmlHttpRequestConnection.js

```diff
@@ -54,6 +64,26 @@ define(["structures/Response", "structures/CAPIError"], function (Response, CAPI
             method = "POST";
         }
 
+        if (xhrCallbacks && Object.keys(xhrCallbacks).length) {
+            if (xhrCallbacks.upload && Object.keys(xhrCallbacks.upload).length) {
+                _assignCallback(XHR.upload, 'onloadstart', xhrCallbacks.upload.onloadstart);
```

Comment:
reading that, I realize, it would be a bit nicer to pass the event handlers with the event name instead of the `on*` syntax, like:
```js
{
    'progress': trackProgress,
    // instead of onprogress
}
```

and then here to write, something like:
```js
Object.keys(eventHandlers.upload).forEach(function (evtName) {
    XHR.upload.addEventListener(evtName, eventHandlers.upload[evtName]);
});
Object.keys(eventHandlers).forEach(function (evtName) {
    if ( evtName !== 'upload' ) {
        XHR.addEventListener(evtName, eventHandlers.upload[evtName]);
    }
});
```
it's up to the developer to use existing event name and to make sure a function is passed.